### PR TITLE
Feature - add capability to create folder in Azure file share

### LINKED ISSUE #103: Azure File Share: Provide Script to create folders within a File Share
**Is your feature request related to a problem? Please describe.**
Currently, this is not supported through ARM-templates, hence the idea to create a PS-script for this being part of the **Arcus.Scripting.Storage.FileShare**

**Describe the solution you'd like**
Something similar to this:
https://github.com/mbraekman/Powershell/blob/master/Azure/AzureStorage-CreateFolderWithinAzureFileShare.ps1

Suggested command name (open to better suggestions):
- _Create-AzFileShareStorageFolder_

**Describe alternatives you've considered**
Since ARM-templates do not support this, the alternative is to create it manually.

### PR DESCRIPTION
**Pull request by stijnmoreels**
Creates a new PS module script to add a folder to a Azure file share resource.

Closes #103

### CODE REVIEW

File: docs/preview/features/powershell/azure-storage-fileshare.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+---
+title: "Scripts related to interacting with Azure Storage for file shares"
+layout: default
+---
+
+# Azure Storage for file shares
+
+This module provides the following capabilities:
+- [Creating a new folder on a file share](#creating-a-folder-on-a-file-share)
```
Comment (mbraekman) [inline docs/preview/features/powershell/azure-storage-fileshare.md:9]:
```suggestion
- [Creating a new folder on an Azure file share](#creating-a-folder-on-an-azure-file-share)
```

File: docs/preview/features/powershell/azure-storage-fileshare.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+## Installation
+
+To have access to the following features, you have to import the module:
+
+```powershell
+PS> Install-Module -Name Arcus.Scripting.Storage.FileShare
+```
+
+## Creating a folder on a file share
```
Comment (mbraekman) [inline docs/preview/features/powershell/azure-storage-fileshare.md:19]:
```suggestion
## Creating a folder on an Azure file share
```

File: docs/preview/features/powershell/azure-storage-fileshare.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+Creates a new folder within the Azure Store File Share resource.
```
Comment (mbraekman) [inline docs/preview/features/powershell/azure-storage-fileshare.md:21]:
```suggestion
Creates a new folder within the Azure File Share resource.
```

File: docs/preview/features/powershell/azure-storage-fileshare.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+| Parameter            | Mandatory | Description                                                             |
+| -------------------- | --------- | ----------------------------------------------------------------------- |
+| `ResourceGroupName`  | yes       | The resource group containing the Azure File Share.                     |
+| `StorageAccountName` | yes       | The Azure Storage account name that has access to the Azure File Share. |
```
Comment (mbraekman) [inline docs/preview/features/powershell/azure-storage-fileshare.md:26]:
```suggestion
| `StorageAccountName` | yes       | The Azure Storage Account name that hosting the Azure File Share. |
```

File: docs/preview/features/powershell/azure-storage-table.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
+---
+title: "Scripts related to interacting with Azure Storage"
```
Comment (mbraekman) [inline docs/preview/features/powershell/azure-storage-table.md:2]:
```suggestion
title: "Scripts related to interacting with Azure Table Storage"
```

File: docs/preview/features/powershell/azure-storage-table.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+layout: default
+---
+
+# Azure Storage
```
Comment (mbraekman) [inline docs/preview/features/powershell/azure-storage-table.md:6]:
```suggestion
# Azure Storage - Tables
```

File: docs/preview/features/powershell/azure-storage.md [hunk 7]
```diff
@@ -1,9 +1,9 @@
 ---
-title: "Scripts related to interacting with Azure Storage"
+title: "Scripts related to interacting with Azure Table Storage"
```
Comment (mbraekman) [inline docs/preview/features/powershell/azure-storage.md:2]: This file will be the "overview"-page for all storage-related modules. (related to #67).
I would use this page to provide an overview of the listed modules and explain how to install the meta package `Arcus.Scripting.Storage` (once it exists)

```suggestion
title: "Scripts related to interacting with Azure Storage"
```

The section below would look like this:
 # Azure Table Storage
  This module provides the following capabilities:
  ... -> link to the table storage-documentation page to the correct section.

 # Azure File Share
  This module provides the following capabilities:
  ... -> link to the table storage-documentation page to the correct section.

File: docs/preview/index.md [hunk 8]
```diff
@@ -25,7 +25,8 @@ For more granular packages we recommend reading the documentation.
 * Automate Azure Logic Apps tasks ([powershell](features/powershell/azure-logic-apps))
 * Automate Azure Resource Manager (ARM) tasks ([powershell](features/powershell/arm))
 * Automate Azure Security tasks ([powershell](features/powershell/azure-security))
-* Automate Azure Storage tasks ([powershell](features/powershell/azure-storage))
+* Automate Azure Storage (File Share) tasks ([powershell](features/powershell/azure-storage-fileshare))
+* Automate Azure Storage (Table) tasks ([powershell](features/powershell/azure-storage-table))
```
Comment (mbraekman) [inline docs/preview/index.md:29]: Over here we should then link to the metapackage-documentation page (Arcus.Scripting.Storage), which links on towards the Table/FileShare-specific documentation pages.

File: src/Arcus.Scripting.Storage.FileShare/Arcus.Scripting.Storage.FileShare.psm1 [hunk 9]
```diff
@@ -0,0 +1,31 @@
+<#
+ .Synopsis
+  Creates a folder within a Azure File Share.
+  
+ .Description
+  Creates a new folder within the Azure Store File Share resource.
```
Comment (mbraekman) [inline src/Arcus.Scripting.Storage.FileShare/Arcus.Scripting.Storage.FileShare.psm1:6]:
```suggestion
  Creates a new folder within the Azure File Share resource.
```

File: src/Arcus.Scripting.Storage.FileShare/Arcus.Scripting.Storage.FileShare.psm1 [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+
+ .Parameter ResourceGroupName
+  The resource group containing the Azure File Share.
+  
+ .Parameter StorageAccountName
+  The Azure Storage account name that has access to the Azure File Share.
```
Comment (mbraekman) [inline src/Arcus.Scripting.Storage.FileShare/Arcus.Scripting.Storage.FileShare.psm1:12]:
```suggestion
  The Azure Storage account name that is hosting the Azure File Share.
```

**Review by mbraekman — COMMENTED** [review]
Some suggestions on the documentation + on how to prepare it already for when the meta package 'Arcus.Scripting.Storage' would be created.
Or perhaps that would be something for a next/seperate PR, specifically for #67

File: docs/preview/features/powershell/azure-storage.md (hunk 7 above)
Comment (stijnmoreels) [reply docs/preview/features/powershell/azure-storage.md]: I updated this page with a menu like you suggested; also with the blob storage I got from `master`.

File: docs/preview/index.md (hunk 8 above)
Comment (stijnmoreels) [reply docs/preview/index.md:29]: Yes, did that. Maybe here also quick-link to the other storages to find directly the one you seek?

Comment (mbraekman) [reply docs/preview/index.md]: Yes, good idea!

**Review by mbraekman — APPROVED** [review]
Looks good!

**Pull request merged**